### PR TITLE
WIP Added Dataprep Base Hook, Dataprep Hook, Operator and docs

### PR DESCRIPTION
**Pull request by michalslowikowski00**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
What is the difference between GoogleDataprepHook and DataprepBaseHook? I think one hook is enough.  Do you expect you to need to add more Hooks?

File: airflow/providers/google/cloud/hooks/dataprep.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from airflow.providers.google.common.hooks.base_dataprep import DataprepBaseHook
+
+
+class GoogleDataprepHook(DataprepBaseHook):
+    """
+    Hook for connection with Dataprep API.
+
+    """
+
+    @retry(stop=stop_after_attempt(5))
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataprep.py:34]: Should this value be configurable?

File: airflow/providers/google/cloud/hooks/dataprep.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def get_jobs_for_job_group(self, job_id: int):
+        """
+        Get information about the batch jobs within a Cloud Dataprep job.
+
+        :param job_id The ID of the job that will be fetched.
+        :type job_id: int
+        """
+        url: str = self.url + f"{job_id}/jobs"
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataprep.py:42]:
```suggestion
        url: str = f"{self.url}{job_id}/jobs"
```

Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/dataprep.py:42]: I would skip `/` from the `self.url` so I can make `f"{self.url}/{job_id}/jobs"`

File: airflow/providers/google/cloud/operators/dataprep.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(
+        self, job_id: int, dataprep_conn_id: str = "dataprep_conn_id", *args, **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.dataprep_conn_id = dataprep_conn_id
+        self.job_id = job_id
+
+    def execute(self, context: Dict):
+        hook = GoogleDataprepHook(dataprep_conn_id=self.dataprep_conn_id)
+        hook.get_jobs_for_job_group(job_id=self.job_id)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/operators/dataprep.py:53]: > Get information about the batch jobs within a Cloud Dataprep job.

Suggest that a value should be returned?

File: airflow/providers/google/common/hooks/base_dataprep.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    :param dataprep_conn_id: The connection ID to use when fetching connection info.
+    :type dataprep_conn_id: str
+    """
+
+    token = os.environ["DATAPREP_TOKEN"]
+    URL = "https://api.clouddataprep.com/v4/jobGroups/"
+
+    def __init__(self, dataprep_conn_id: str = "dataprep_conn_id") -> None:
+        super().__init__()
```
Comment (turbaszek) [inline airflow/providers/google/common/hooks/base_dataprep.py:49]: As @mik-laj said - I'm do not understand why we need base hook for dataprep

**Comment by michalslowikowski00** [comment]
@mik-laj and @turbaszek I remember when I ask Kamil if I should create a base hook and he said yes. I thought that the Dataprep is different service than Google and requires its own connection like Google. But ok, I will remove Base Hook.

File: airflow/providers/google/cloud/operators/dataprep.py (hunk 3 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/operators/dataprep.py:53]: Should be and will be.

File: airflow/providers/google/cloud/hooks/dataprep.py (hunk 2 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/dataprep.py:42]: Good one, thx. :)

File: airflow/providers/google/cloud/hooks/dataprep.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        @retry(stop=stop_after_attempt(self.retry_no))
+        def _do_task():
+
+            url: str = f"{self.url}{job_id}/jobs"
+
+            response = requests.get(url, headers=self.headers)
+            response.raise_for_status()
+
+        return _do_task()
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataprep.py:60]: How about more generalization:
```python
def get(self, *args, **kwargs):
    def _get(*args, **kwargs):
        resp = requests.get(*args, **kwargs)
        resp.raise_for_status()
        return resp
    return retry(stop=stop_after_attempt(self.retry_no))(_get(*args, **kwargs))
```

File: airflow/providers/google/common/hooks/base_dataprep.py (hunk 4 above)
Comment (michalslowikowski00) [reply airflow/providers/google/common/hooks/base_dataprep.py]: Fixed.

File: airflow/providers/google/cloud/hooks/dataprep.py (hunk 1 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/dataprep.py]: under construction. :)

File: airflow/providers/google/cloud/hooks/dataprep.py (hunk 5 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/dataprep.py:60]: Nice one, I like it.

Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/dataprep.py:60]:
```python
```python
def get(self, *args, **kwargs):
     @retry(stop=stop_after_attempt(self.retry_no))
    def _get(*args, **kwargs):
        resp = requests.get(*args, **kwargs)
        resp.raise_for_status()
        return resp
    return _get(*args, **kwargs)
```

**Pull request closed without merging**